orchestrator/cli/migrate_workflows.py

```diff
@@ -0,0 +1,230 @@
+    print_fmt("\nDelete existing workflow\n", flags=[COLOR.UNDERLINE])
+    already_used_workflows = {wf["name"] for wf in state["workflows_to_add"] + state["workflows_to_delete"]}
+    items = [
+        (wf.name, wf.target, wf.description, wf.products[0].product_type)
+        for wf in workflows
+        if wf.products and wf.name not in already_used_workflows
+    ]
+    keys = ["#", "name", "target", "description", "product_type"]
+    if not items:
+        _noqa_print("No deletable workflows in database")
+        return state
+
+    print_fmt(
+        tabulate(
+            items,
+            headers=keys,
+            showindex=itertools.count(1),
+        ),
+        end="\n\n",
+    )
+    wf_num = get_user_input("Which workflow do you want to delete? (q to cancel) ", "q")
+    if not wf_num.isdigit():
+        return state
+    wf_index = int(wf_num) - 1
+    if 0 <= wf_index < len(items):
+        item = dict(zip(keys[1:], items[wf_index]))
+        return {**state, "workflows_to_delete": [*state["workflows_to_delete"], item]}
+    else:
+        return state
+
+
+def _show_state(state: dict) -> dict:
+    print_fmt("\nWorkflows to add:", flags=[COLOR.GREEN])
+    print_fmt(
+        tabulate(
+            state["workflows_to_add"],
+            headers="keys",
+            showindex=itertools.count(1),
+        ),
+        end="\n\n",
+    )
+
+    print_fmt("Workflows to delete:", flags=[COLOR.RED])
+    print_fmt(
+        tabulate(
+            state["workflows_to_delete"],
+            headers="keys",
+            showindex=itertools.count(1),
+        ),
+        end="\n\n",
+    )
+    return state
+
+
+def delete_dangling_workflows(workflows: List[WorkflowTable], state: dict) -> dict:
+    print_fmt(
+        "\nThe following workflows were found in the database that do not have a corresponding LazyWorkflowInstance:\n"
+    )
+    # All workflow products have the same product_type, so we take the first product
+    # Workflows without a product  will not be listed
+    _print_workflows_table(workflows)
+    should_delete_dangling_workflows = (
+        get_user_input("Do you wish to delete all dangling workflows from the database? [y/n]: ", "n").lower() == "y"
+    )
+
+    if not should_delete_dangling_workflows:
+        _noqa_print("Cancelling")
+        return state
+
+    already_used_workflows = {wf["name"] for wf in state["workflows_to_add"] + state["workflows_to_delete"]}
+    keys = ["name", "target", "description", "product_type"]
+    items = [
+        {
+            "name": wf.name,
+            "target": wf.target,
+            "description": wf.description,
+            "product_type": wf.products[0].product_type,
+        }
+        for k, wf in zip(keys, workflows)
+        if wf.products and wf.name not in already_used_workflows
+    ]
+    return {**state, "workflows_to_delete": [*state["workflows_to_delete"], *items]}
+
+
+def create_workflows_migration_wizard() -> Tuple[List[dict], List[dict]]:
```

Comment:
It would be good to show which workflows are not used yet from `LazyWorkflowInstance`.

maybe at the top which workflows aren't in the database or at each workflow a `<- used` or `<- unused`

Reply:
Added to followup improvement issue